src/messengers/vk/vk/updates.go

```diff
@@ -0,0 +1,226 @@
+}
+
+var errCommandNotFound = fmt.Errorf("command not found")
+
+func (m *Messenger) processCommand(message object.MessagesMessage, chat *msg.Chat) error {
+	var err error
+	if strings.HasPrefix(message.Text, "/get_token") {
+		err = m.processGetToken(message, chat)
+	} else if strings.HasPrefix(message.Text, "/subscribe") {
+		err = m.processSubscribe(message, chat)
+	} else if strings.HasPrefix(message.Text, "/unsubscribe") {
+		err = m.processUnsubscribe(message, chat)
+	} else {
+		return errCommandNotFound
+	}
+	return m.processCommandResult(err, chat)
+}
+
+// processCommandResult checks if there is an error that needs to be sent to the user and tries to send it.
+// If the error was internal, it is added to the returned error
+//
+// It might have some messenger-specific logic in the future, so it should not be moved to baseMessenger.
+func (m *Messenger) processCommandResult(err error, chat *msg.Chat) error {
+	// If err is nil or not a grpc error, it should be returned immediately
+	if status.Code(err) == codes.OK {
+		return err
+	}
+	_, sendErr := m.SendMessage(context.TODO(), &msg.SendMessageRequest{
+		Message: &msg.Message{Text: status.Convert(err).Message()},
+		Chat:    chat,
+	})
+	if !messenger.IsUserInputError(err) {
+		if sendErr != nil {
+			return fmt.Errorf("could not process command: %v, could send error %v", err, sendErr)
+		}
+		return err
+	}
+	return sendErr
+}
+
+func (m *Messenger) processGetToken(_ object.MessagesMessage, chat *msg.Chat) error {
+	token, err := m.GetChatToken(chat.Id, chat.Type)
+	if err != nil {
+		return err
+	}
+	_, errSend := m.SendMessage(context.TODO(), &msg.SendMessageRequest{
+		Message: &msg.Message{Text: token},
+		Chat:    chat,
+	})
+	return errSend
+}
+
+func (m *Messenger) processSubscribe(message object.MessagesMessage, chat *msg.Chat) error {
+	s := strings.Split(message.Text, " ")
+	return m.SubscribeCallback(chat, s[len(s)-1])
+}
+
+func (m *Messenger) processUnsubscribe(message object.MessagesMessage, chat *msg.Chat) error {
+	s := strings.Split(message.Text, " ")
+	return m.UnsubscribeCallback(chat, s[len(s)-1])
+}
+
+func (m *Messenger) processWall(wall object.WallWallpost, chat *msg.Chat) error {
+	message := msg.Message{
+		Text: wall.Text,
+		Sender: &msg.Sender{
+			Name: m.getWallAuthor(&wall),
+			Chat: &msg.Chat{Name: "vk"},
+		},
+	}
+
+	for _, attachment := range wall.Attachments {
+		if attachment.Type == "photo" {
+			message.Attachments = m.processPhoto(attachment.Photo, chat.Id, message.Attachments)
+		}
+	}
+	return m.MessageCallback(&message, chat)
+}
+
+func (m *Messenger) getWallAuthor(wall *object.WallWallpost) string {
+	name := ""
+	if wall.FromID > 0 { // user
+		userResponse, err := m.vk.UsersGet(api.Params{"user_ids": wall.FromID})
+		if err != nil || len(userResponse) == 0 {
+			log.Print("could not find user with id ", wall.FromID)
+			return ""
+		}
+		name = userResponse[0].FirstName + " " + userResponse[0].LastName
```

Comment:
there should be some separate function like ```getUserNameByID```, because this piece of code is too similar to ```getSenderName``` function